### DISCUSSION TITLE
`:health` command to get health of currently active language

### DISCUSSION THREAD

**Discussion by A-Walrus** [helix-editor/helix#10216] [Ideas]
I find myself sometimes, opening a file in a language I don't frequently use, and I'm not sure weather I have an LSP and everything setup correctly, would be nice to be able to run `:health` and get the status for whatever language is currently active, without having to go to a shell and run `hx --health <lang>`

**Comment by alphastrata** [comment]
What do you imagine the feedback to be? something in the status line? a popup? a readout in a new scratch buffer like neovim does?

**Comment by A-Walrus** [comment]
I was thinking status line or pop-up. Status line seems too small for the amount of text we want to show, so I guess pop up makes sense

**Comment by A-Walrus** [comment]
The theoretical plugin script you showed wouldn't work since the language is not the same as the extension:
`.js` = `javascript`
`.rs` = `rust`
`.pl` = `perl` or `prolog`

**Comment by kirawi** [comment]
@ghc0 Please try to use respectful language in the future to keep discussion healthy.

**Comment by kirawi** [comment]
Again, please use respectful language and avoid making assumptions about individuals and their beliefs. While there isn't an agreed upon code of conduct for this project, it follows along the general rules of [Rust's code of conduct](https://www.rust-lang.org/policies/code-of-conduct) and [Microsoft's code of conduct](https://microsoft.github.io/codeofconduct/).

Different perspectives and opinions on an issue are appreciated, as long as they are conveyed in a respectful and constructive manner.
